### PR TITLE
Processor identifier to centralProcessor

### PR DESCRIPTION
**Pull request by Praveen101997**
Move the ProcessorIdentifier inner class to the CentralProcessor class

### CODE REVIEW

**Comment by Praveen101997** [comment]
> Move the ProcessorIdentifier inner class to the CentralProcessor class
@dbwiddis 
I commit the changes as per requirement
But not sure as it is my first contribution
If anything is wrong, please guide and correct me

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java [hunk 1]
```diff
@@ -36,38 +36,38 @@
...
-    String getVendor();
+    @Deprecated String getVendor();
 
     /**
      * Name, eg. Intel(R) Core(TM)2 Duo CPU T7300 @ 2.00GHz
      *
      * @return Processor name.
      */
-    String getName();
+    @Deprecated String getName();
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: While technically correct, for style purposes put in a newline after the `@Deprecated`.  Also add a `@deprecated` annotation to the javadoc (note the lower case) with a comment `replaced by {@link ProcessorIdentifier#getName()}`

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java [hunk 2]
```diff
@@ -453,4 +458,84 @@ public int getProcessorGroup() {
...
+            Matcher matcher = pattern.matcher(cpuName);
+            if (matcher.find()) {
+                String unit = matcher.group(1);
+                this.cpuVendorFreq = ParseUtil.parseHertz(unit);
+            } else {
+                this.cpuVendorFreq = -1L;
+            }
+        }
+
+        public String getVendor() {
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: Good, but these need javadocs. You can copy the ones from above that you are deprecating.

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java [hunk 3]
```diff
@@ -339,6 +339,11 @@ public int getIndex() {
         }
     }
 
+
+    ProcessorIdentifier getProcessorIdentifier();
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: You'll need to write a new javadoc for this.  Something like:

> Returns the CPU's identifier strings, including stepping, model, and family information (also called the signature of a CPU)

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java [hunk 4]
```diff
@@ -453,4 +458,84 @@ public int getProcessorGroup() {
...
+        }
+
+        public boolean isCpu64bit() {
+            return cpu64bit;
+        }
+
+        public long getVendorFreq() {
+            return cpuVendorFreq;
+        }
+
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: for completeness, let's add a `toString()` that returns `getIdentifier()`.

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java (hunk 1 above)
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: These are valid, but for style purposes add a new line between the `@Deprecated` annotation and the `get`.  You also need to add the (lowercase) `@deprecated` annotation in the Javadoc, pointing to the new location this information will be.  Such as `@deprecated Replaced by {@link ProcessorIdentifier#getName}`

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java (hunk 3 above)
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: This is good here but you haven't implemented it in the AbstractCentralProcessor class, so you're failing compilation.  Also you need a Javadoc for this, something like 

> Returns the CPU's identifier strings such as the CPU's stepping, model, and family information (also called the signature of a CPU).

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java (hunk 2 above)
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: Good but you need javadocs here.  Copy the ones from the deprecated getters.

File: oshi-core/src/main/java/oshi/hardware/CentralProcessor.java (hunk 4 above)
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/CentralProcessor.java]: Let's add a toString() that returns `getIdentifier()`.

File: oshi-core/src/main/java/oshi/hardware/common/AbstractCentralProcessor.java [hunk 5]
```diff
@@ -98,47 +98,56 @@ public AbstractCentralProcessor() {
 
     @Override
     public String getVendor() {
-        return cpuid.get().cpuVendor;
+        return getProcessorIdentifier().getVendor();
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/common/AbstractCentralProcessor.java]: These are all good, but don't forget to add a few lines above, a `getProcesserIdentifier()` implementation that returns `cpuid.get()`.

**Comment by dbwiddis** [comment]
Also this needs a [CHANGELOG](https://github.com/oshi/oshi/blob/master/CHANGELOG.md) entry.

**Comment by Praveen101997** [comment]
> Also this needs a [CHANGELOG](https://github.com/oshi/oshi/blob/master/CHANGELOG.md) entry.

Do I have to also import this library in CentralProcessor which is previously imported in abstractCentralProcessor
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import oshi.util.ParseUtil;

**Comment by Praveen101997** [comment]
@dbwiddis  I didn't add the issue name in the changelog.
So, I again pushed changes in it.
And the process is in way.

**Pull request merged**